### ISSUE TITLE
Contex menu on tab headers

### ISSUE THREAD

**Issue by pylorak** [pylorak/reloadmatic#1]
Waiting for [Fx Bug #1215376](https://bugzilla.mozilla.org/show_bug.cgi?id=1215376) to be solved, then it will be possible to add this feature.

**Comment by Vampire** [comment]
[This plugin](piro.sakura.ne.jp/xul/_treestyletab.html.en) has a custom menu entry in the tab context menu. Shouldn't it work for ReloadMatic then too? :-) (Sorry if not, I'm not familiar with FF plugin development)

**Comment by pylorak** [comment]
The exact problem is that, even though I can put a context menu in the tab header, I cannot make it different for each tab. ReloadMatic would need to make the context menu vary for each tab based on which options are enabled for the given tab. TreeStyleTab doesn't need that, so menus in the tab header work alright for that addon, but not for ReloadMatic :(

**Comment by Vampire** [comment]
Ah, I see, to see what is currently selected.
Maybe as intermediate solution you can have some generic entry that opens a dialog with the settings for the respective tab or something like that?
If you have some website that prevents right-click, you have no choice to enable auto-reloading currently.

**Comment by pylorak** [comment]
Yeah, issue #3 is tracking exactly that kind of problem. I'd like to wait a tiny bit to see what efforts Mozilla make in closing these API holes, and only resort to custom HTML UIs when necessary.

**Comment by Vampire** [comment]
Ah, ok. Well, from what I read it will be no fun to wait for Mozdevs to do what the add-on devs need. :-D

**Comment by Vampire** [comment]
Hm, you could as an intermediary solution make a tab context menu that has not toggle buttons and no state indicator like "Smart timing on" and "Smart timing off" and so on. This way it would be usable at least, just not as comfortable as it will be after the API extension was implemented.

**Comment by pylorak** [comment]
Yes that would be possible. This is also what some other tab extensions have done. But since ReloadMatic has a "working" menu in the page context that can show state, most user would think the tab menu is buggy, and would be annoyed.

**Comment by Vampire** [comment]
Well, if there are enable and disable buttons and set-to buttons, it should be quite clear that it is different. Besides that, it would make the add-on usable on pages that block or replace the right-click. In my personal opinion this would justify annoying a few users that don't get it right away. :-D

**Comment by syldub69** [comment]
The FoxyTab addon for instance has a context menu integrated into the tab context menu, and allows setting different timer configuration on different tabs.
Couldn't you implement the same type of things ?

**Comment by pylorak** [comment]
The problem is not showing the tab's context menu, but making it vary dynamically based on the tab's state. Please be patient. Even though this feature is not possible right now, the prerequirements in FF are slowly getting implemented, and I'll jump on it as soon as possible. This is on my wishlist too.

**Comment by pylorak** [comment]
This has been implemented, now possible thanks to the new APIs in FF60. **Important:** You need both ReloadMatic 1.7.0 (already published) AND Firefox 60 (not yet released) for this feature to work.
